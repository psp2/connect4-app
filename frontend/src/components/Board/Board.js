import React from "react";
import "./Board.scss";

function Board(props) {
  function attemptMove(row, col) {
    props.makeMove(row, col);
  }

  function displayToken(token, row, col) {
    if (token === 0) {
      return (
        <div onClick={() => attemptMove(row, col)} className="WhiteSpace"></div>
      );
    } else if (token === 1) {
      return <div className="YellowToken"></div>;
    } else if (token === 2) {
      return <div className="RedToken"></div>;
    }
  }

  function displayRow(row, rowIndex) {
    return (
      <div className="BoardRow">
        {row.map((token, colIndex) => displayToken(token, rowIndex, colIndex))}
      </div>
    );
  }

  return (
    <div className="Board">
      {props.gameState &&
        props.gameState.map((row, rowIndex) => displayRow(row, rowIndex))}
    </div>
  );
}

export default Board;
