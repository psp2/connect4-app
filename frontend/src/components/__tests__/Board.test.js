import React from "react";
import Board from "../Board/Board";
import renderer from "react-test-renderer";

it("small board, renders correctly", () => {
  const player = 1;
  const gameState = [
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("medium board, renders correctly", () => {
  const player = 1;
  const gameState = [
    [0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("large board, renders correctly", () => {
  const player = 1;
  const gameState = [
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("null player", () => {
  const player = null;
  const gameState = [
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("null board", () => {
  const player = null;
  const gameState = null;
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("empty board", () => {
  const player = null;
  const gameState = [];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("yellow board, renders correctly", () => {
  const player = 1;
  const gameState = [
    [1, 1, 1, 1, 1, 1],
    [1, 1, 1, 1, 1, 1],
    [1, 1, 1, 1, 1, 1],
    [1, 1, 1, 1, 1, 1],
    [1, 1, 1, 1, 1, 1],
    [1, 1, 1, 1, 1, 1],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("red board, renders correctly", () => {
  const player = 1;
  const gameState = [
    [2, 2, 2, 2, 2, 2],
    [2, 2, 2, 2, 2, 2],
    [2, 2, 2, 2, 2, 2],
    [2, 2, 2, 2, 2, 2],
    [2, 2, 2, 2, 2, 2],
    [2, 2, 2, 2, 2, 2],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("too large board, renders incorrectly", () => {
  const player = 1;
  const gameState = [
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("too small board, renders incorrectly", () => {
  const player = 1;
  const gameState = [
    [0, 0, 0],
    [0, 0, 0],
    [0, 0, 0],
  ];
  const tree = renderer
    .create(<Board player={player} gameState={gameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
