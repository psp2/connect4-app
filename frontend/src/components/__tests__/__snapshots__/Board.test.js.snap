// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty board 1`] = `
<div
  className="Board"
/>
`;

exports[`large board, renders correctly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`medium board, renders correctly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`null board 1`] = `
<div
  className="Board"
/>
`;

exports[`null player 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`red board, renders correctly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
    <div
      className="RedToken"
    />
  </div>
</div>
`;

exports[`small board, renders correctly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`too large board, renders incorrectly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`too small board, renders incorrectly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
    <div
      className="WhiteSpace"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`yellow board, renders correctly 1`] = `
<div
  className="Board"
>
  <div
    className="BoardRow"
  >
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
  </div>
  <div
    className="BoardRow"
  >
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
    <div
      className="YellowToken"
    />
  </div>
</div>
`;
